Add Prometheus TSDB Run-Length-Encoding project idea (#518)
cncf/mentoring: summerofcode/2022.md

- Mentor(s): Sedef Savas (@sedefsavas) Richard Case (@richcase)
- Upstream Issue (URL): https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2178

### Prometheus

> #### Add Rule-Length-Encoding as TSDB encoding

Reviewer: How about this? (And it is Run for R)
```suggestion
#### Add Run-Length-Encoding for samples in TSDB
```

Author: lol, that has to be one of my best typos in a long time. :joy:

> - Description: Prometheus stores the scraped data in it's own [Time Series Database](https://www.wikiwand.com/en/Time_series_database) (TSDB). Due to the nature of often collecting simliar values they can be stored in sequenence using optimized encodings. In Prometheus v2.0 and above all values are encoded using [XOR'ed](https://blog.acolyer.org/2016/05/03/gorilla-a-fast-scalable-in-memory-time-series-database/) int64. Almost since the initial v2.0 release there is a long standing issue on adding other encodings to Prometheus' TSDB. Especially cases where either all values of the series are either `0` or `1` are prime candidates for [Run-Length-Encoding](https://www.wikiwand.com/en/Run-length_encoding) (RLE). After conducting an [experiment on the real world feasbility for adding RLE](https://github.com/prometheus/prometheus/pull/10275), that looks very promising (saving up to 10% disk), we want to work with you to improve Prometheus' storage!

Reviewer: ```suggestion
- Description: Prometheus stores the scraped data in it's own [Time Series Database](https://www.wikiwand.com/en/Time_series_database) (TSDB). Due to the nature of often collecting simliar values they can be stored in sequenence using optimized encodings. In Prometheus v2.0 and above all values are encoded using [XOR'ed](https://blog.acolyer.org/2016/05/03/gorilla-a-fast-scalable-in-memory-time-series-database/) int64. Almost since the initial v2.0 release there is a long standing issue on adding other encodings to Prometheus' TSDB. Especially cases where either all values of the series are either `0` or `1` are prime candidates for [Run-Length-Encoding](https://www.wikiwand.com/en/Run-length_encoding) (RLE). After conducting an [experiment on the real world feasbility for adding RLE](https://github.com/prometheus/prometheus/pull/10275), that looks very promising (saving up to 10% disk). We want to work with you to implement RLE in TSDB improve Prometheus' storage!
```

> - Description: Prometheus stores the scraped data in it's own [Time Series Database](https://www.wikiwand.com/en/Time_series_database) (TSDB). Due to the nature of often collecting simliar values they can be stored in sequenence using optimized encodings. In Prometheus v2.0 and above all values are encoded using [XOR'ed](https://blog.acolyer.org/2016/05/03/gorilla-a-fast-scalable-in-memory-time-series-database/) int64. Almost since the initial v2.0 release there is a long standing issue on adding other encodings to Prometheus' TSDB. Especially cases where either all values of the series are either `0` or `1` are prime candidates for [Run-Length-Encoding](https://www.wikiwand.com/en/Run-length_encoding) (RLE). After conducting an [experiment on the real world feasbility for adding RLE](https://github.com/prometheus/prometheus/pull/10275), that looks very promising (saving up to 10% disk), we want to work with you to implement RLE in TSDB and improve Prometheus' storage!

Author: ```suggestion
- Description: Prometheus stores the scraped data in it's own [Time Series Database](https://en.wikipedia.org/wiki/Time_series_database) (TSDB). Due to the nature of often collecting simliar values they can be stored in sequenence using optimized encodings. In Prometheus v2.0 and above all values are encoded using [XOR'ed](https://blog.acolyer.org/2016/05/03/gorilla-a-fast-scalable-in-memory-time-series-database/) int64. Almost since the initial v2.0 release there is a long standing issue on adding other encodings to Prometheus' TSDB. Especially cases where either all values of the series are either `0` or `1` are prime candidates for [Run-Length-Encoding](https://en.wikipedia.org/wiki/Run-length_encoding) (RLE). After conducting an [experiment on the real world feasbility for adding RLE](https://github.com/prometheus/prometheus/pull/10275), that looks very promising (saving up to 10% disk), we want to work with you to implement RLE in TSDB and improve Prometheus' storage!
```

Author: Seems like I slipped some wikiwand rather than wikipedia :upside_down_face:
